# SHORT LIST OF CARM_VIS OBJECTS

This small notebook just gets and stores a short list of the CARM_VIS objets, making two subsamples:

- The first one choosing those with the choosing only those that have correct GLS periodograms calculated. 
- For the second subsample, choose the RV curves with the lower RMS values (no matter if they have the GLS periodograms correctly calculated or not).


## Modules and configuration

### Modules

In [15]:
import pandas as pd
import numpy as np

### Configuration

In [2]:
INPUT_FILE = "../data/GTO_objects_with_PG.csv"
OUTPUT_FILE = "../data/SELECTION_GTO_objects_with_PG.csv"

##  Load the GTO objects

In [3]:
df = pd.read_csv(INPUT_FILE, sep=',', decimal='.')
df.head()

Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
0  J23585+076        Wolf 1051   AB  4383    359.636420      7.656947   
1  J23556-061           GJ 912   AB   912    358.913617     -6.144283   
2  J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
3  J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
4  J23492+024           BR Psc    -   908    357.306604      2.396918   

      RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ...   T0_PG_TESS  \
0  23:58:32.65  +07:39:30.1   100.839686   -52.931049  ...          NaN   
1  23:55:39.78  -06:08:33.4    88.129933   -65.175491  ...          NaN   
2  23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...  1764.609498   
3  23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...  1354.108815   
4  23:49:12.53  +02:24:04.4    93.567467   -56.885396  ...          NaN   

  e_T0_PG_TESS  offset_PG_TESS e_offset_PG_TESS  FAP_PG_TESS  valid_PG_TESS  \
0          NaN             NaN              NaN          NaN            0.0   
1          NaN             NaN              NaN          NaN            0.0   
2     0.010704     8262.219751         1.365667     1.000000            1.0   
3     0.001261     7767.134654         0.094298     0.064148            1.0   
4          NaN             NaN              NaN          NaN            0.0   

                                       error_PG_TESS  elapsed_time_PG_TESS  \
0  Not recognized as a supported data product:\nn...              0.002199   
1  Not recognized as a supported data product:\nn...              0.000974   
2                                                NaN            344.002685   
3                                                NaN            473.533042   
4  Not recognized as a supported data product:\nn...              0.000984   

                                      fits_file_TESS  \
0                                                NaN   
1                                                NaN   
2  ../data/CARMENES_GTO_TESS_PGs/J23548+385_TESS_...   
3  ../data/CARMENES_GTO_TESS_PGs/J23505-095_TESS_...   
4                                                NaN   

                                       fig_file_TESS  
0                                                NaN  
1                                                NaN  
2  ../data/CARMENES_GTO_TESS_PGs/figures/J23548+3...  
3  ../data/CARMENES_GTO_TESS_PGs/figures/J23505-0...  
4                                                NaN  

[5 rows x 248 columns]

In [9]:
print(list(df.columns))

['Karmn', 'Name', 'Comp', 'GJ', 'RA_J2016_deg', 'DE_J2016_deg', 'RA_J2000', 'DE_J2000', 'l_J2016_deg', 'b_J2016_deg', 'Ref01', 'SpT', 'SpTnum', 'Ref02', 'Teff_K', 'eTeff_K', 'logg', 'elogg', '[Fe/H]', 'e[Fe/H]', 'Ref03', 'L_Lsol', 'eL_Lsol', 'Ref04', 'R_Rsol', 'eR_Rsol', 'Ref05', 'M_Msol', 'eM_Msol', 'Ref06', 'muRA_masa-1', 'emuRA_masa-1', 'muDE_masa-1', 'emuDE_masa-1', 'Ref07', 'pi_mas', 'epi_mas', 'Ref08', 'd_pc', 'ed_pc', 'Ref09', 'Vr_kms-1', 'eVr_kms-1', 'Ref10', 'ruwe', 'Ref11', 'U_kms-1', 'eU_kms-1', 'V_kms-1', 'eV_kms-1', 'W_kms-1', 'eW_kms-1', 'Ref12', 'sa_m/s/a', 'esa_m/s/a', 'Ref13', 'SKG', 'Ref14', 'SKG_lit', 'Ref14_lit', 'Pop', 'Ref15', 'vsini_flag', 'vsini_kms-1', 'evsini_kms-1', 'Ref16', 'P_d', 'eP_d', 'Ref17', 'pEWHalpha_A', 'epEWHalpha_A', 'Ref18', 'log(LHalpha/Lbol)', 'elog(LHalpha/Lbol)', 'Ref19', '1RXS', 'CRT_s-1', 'eCRT_s-1', 'HR1', 'eHR1', 'HR2', 'eHR2', 'Flux_X_E-13_ergcm-2s-1', 'eFlux_X_E-13_ergcm-2s-1', 'LX/LJ', 'eLX/LJ', 'Ref20', 'Activity', 'Ref21', 'FUV_mag',

In [13]:
df['lc_file']

0                                                    NaN
1                                                    NaN
2      ../data/CARMENES_GTO_TESS_lc/lightcurves/J2354...
3      ../data/CARMENES_GTO_TESS_lc/lightcurves/J2350...
4                                                    NaN
                             ...                        
358    ../data/CARMENES_GTO_TESS_lc/lightcurves/J0028...
359                                                  NaN
360    ../data/CARMENES_GTO_TESS_lc/lightcurves/J0018...
361                                                  NaN
362    ../data/CARMENES_GTO_TESS_lc/lightcurves/J0005...
Name: lc_file, Length: 363, dtype: object

Number of GTO objects without an RV file name:

In [19]:
sum(df['rv_file'].isna())

0

Number of GTO objects without an LC file name:

In [20]:
sum(df['lc_file'].isna())

110

## Select the GTO object with valid PG calculated

Total number of GTO objects in the original table:

In [5]:
len(df)

363

Number of GTO objects with correctly calculated periodogram for CARMENES RV curves:

In [7]:
len(df[df['valid_PG_RV'] == 1.0])

355

Number of GTO objects with correctly calculated periodogram for TESS LC:

In [8]:
len(df[df['valid_PG_TESS'] == 1.0])

246

Combined number of GTO objects with correctly calculated periodograms for both CARMENES RV curves and TESS LC:

In [21]:
len(df[(df['valid_PG_RV'] == 1.0) & (df['valid_PG_TESS'] == 1.0)])

239

These 239 object will be the main target of the analysis:

In [22]:
selected_df = df[(df['valid_PG_RV'] == 1.0) & (df['valid_PG_TESS'] == 1.0)].copy()
selected_df.head()

Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
2   J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
3   J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
5   J23431+365          GJ 1289    -  1289    355.781509     36.536310   
7   J23381-162        G 273-093    -  4352    354.532687    -16.236514   
10  J23245+578       BD+57 2735    -   895    351.126628     57.853057   

       RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ...   T0_PG_TESS  \
2   23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...  1764.609498   
3   23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...  1354.108815   
5   23:43:06.31  +36:32:13.1   107.922839   -24.336479  ...  1764.717539   
7   23:38:08.16  -16:14:10.2    61.845437   -69.825220  ...  1354.111098   
10  23:24:30.51  +57:51:15.5   111.552287    -3.085183  ...  1955.800582   

   e_T0_PG_TESS  offset_PG_TESS e_offset_PG_TESS  FAP_PG_TESS  valid_PG_TESS  \
2      0.010704     8262.219751         1.365667     1.000000            1.0   
3      0.001261     7767.134654         0.094298     0.064148            1.0   
5      0.003720    16158.288258         0.164698     0.002785            1.0   
7      0.000422    30353.147900         0.175123     0.031223            1.0   
10     0.001420    84823.865767         0.391298     0.799167            1.0   

    error_PG_TESS  elapsed_time_PG_TESS  \
2             NaN            344.002685   
3             NaN            473.533042   
5             NaN            352.262793   
7             NaN            485.008036   
10            NaN            476.798646   

                                       fits_file_TESS  \
2   ../data/CARMENES_GTO_TESS_PGs/J23548+385_TESS_...   
3   ../data/CARMENES_GTO_TESS_PGs/J23505-095_TESS_...   
5   ../data/CARMENES_GTO_TESS_PGs/J23431+365_TESS_...   
7   ../data/CARMENES_GTO_TESS_PGs/J23381-162_TESS_...   
10  ../data/CARMENES_GTO_TESS_PGs/J23245+578_TESS_...   

                                        fig_file_TESS  
2   ../data/CARMENES_GTO_TESS_PGs/figures/J23548+3...  
3   ../data/CARMENES_GTO_TESS_PGs/figures/J23505-0...  
5   ../data/CARMENES_GTO_TESS_PGs/figures/J23431+3...  
7   ../data/CARMENES_GTO_TESS_PGs/figures/J23381-1...  
10  ../data/CARMENES_GTO_TESS_PGs/figures/J23245+5...  

[5 rows x 248 columns]

## Store the filtered results

In [23]:
selected_df.to_csv(OUTPUT_FILE, sep=',', decimal='.', index=False)

# Summary

**OBSERVATIONS AND CONCLUSIONS:**
- We have 239 GTO objects for which we were able to correctly calculate the periodograms for both CARMENES RV curves and TESS Light Curves.
- These 239 objects will be the source data to compare the results obtained for both types of curves, and also for further analysis.